src/tlo/analysis/utils.py

```diff
@@ -306,43 +306,90 @@ def generate_series(dataframe: pd.DataFrame) -> pd.Series:
+    :Param: only_central: Whether to only report the central value (dropping the range).
+
     """
+    stats = dict()
+
+    if central_measure == 'mean':
+        stats.update({'central': results.groupby(axis=1, by='draw', sort=False).mean()})
+    elif central_measure == 'median':
+        stats.update({'central': results.groupby(axis=1, by='draw', sort=False).median()})
+    else:
+        raise ValueError(f"Unknown stat: {central_measure}")
 
-    summary = pd.concat(
+    stats.update(
         {
-            'mean': results.groupby(axis=1, by='draw', sort=False).mean(),
-            'lower': results.groupby(axis=1, by='draw', sort=False).quantile(0.025),
-            'upper': results.groupby(axis=1, by='draw', sort=False).quantile(0.975),
-        },
-        axis=1
+            'lower': results.groupby(axis=1, by='draw', sort=False).quantile((1.-width_of_range)/2.),
+            'upper': results.groupby(axis=1, by='draw', sort=False).quantile(1.-(1.-width_of_range)/2.),
+        }
     )
+
+    summary = pd.concat(stats, axis=1)
     summary.columns = summary.columns.swaplevel(1, 0)
     summary.columns.names = ['draw', 'stat']
-    summary = summary.sort_index(axis=1)
+    summary = summary.sort_index(axis=1).reindex(columns=['lower', 'central', 'upper'], level=1)
 
-    if only_mean and (not collapse_columns):
+    if only_central and (not collapse_columns):
         # Remove other metrics and simplify if 'only_mean' across runs for each draw is required:
-        om: pd.DataFrame = summary.loc[:, (slice(None), "mean")]
+        om: pd.DataFrame = summary.loc[:, (slice(None), "central")]
         om.columns = [c[0] for c in om.columns.to_flat_index()]
         om.columns.name = 'draw'
         return om
 
     elif collapse_columns and (len(summary.columns.levels[0]) == 1):
         # With 'collapse_columns', if number of draws is 1, then collapse columns multi-index:
         summary_droppedlevel = summary.droplevel('draw', axis=1)
-        if only_mean:
-            return summary_droppedlevel['mean']
+        if only_central:
+            return summary_droppedlevel['central']
         else:
             return summary_droppedlevel
 
     else:
         return summary
 
 
+def summarize(
+    results: pd.DataFrame,
+    only_mean: bool = False,
+    collapse_columns: bool = False
+):
+    """Utility function to compute summary statistics
+
+    Finds mean value and 95% interval across the runs for each draw.
+
+    NOTE: This provides the legacy functionality of `summarize` that is hard-wired to use `means` (the kwarg is
+     `only_mean` and the name of the column in the output is `mean`). Please move to using the new and more flexible
+     version of `summarize` that allows the use of medians and is flexible to allow other forms of summary measure in
+     the future.
+    """
+    output = summarise(
+        results=results,
+        central_measure='mean',
+        only_central=only_mean,
```

Comment:
passing `only_mean` to `only_central` here looks confusing. can't we use one name for consistency?

Reply:
I was aiming to preserve the signature and functionality of 'summarize', so not to disturb existing analysis code. But, to use the new function "behind the scenes" to avoid having to duplicate code. It's not beautiful but the "users" won't see it.